src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateMarcAuthoritiesEventHandler.java

```diff
@@ -0,0 +1,156 @@
+        })
+        .onFailure(e -> {
+          LOGGER.error("Failed to save new Authority", e);
+          future.completeExceptionally(e);
+        });
+    } catch (Exception e) {
+      LOGGER.error("Failed to save new Authority", e);
+      future.completeExceptionally(e);
+    }
+    return future;
+  }
+
+  private Future<Authority> addAuthority(Authority authority, AuthorityCollection authorityCollection) {
+    Promise<Authority> promise = Promise.promise();
+    authorityCollection.add(authority, success -> promise.complete(success.getResult()),
+        failure -> {
+          LOGGER.error(format("Error posting an Authority cause %s, status code %s", failure.getReason(), failure.getStatusCode()));
+          promise.fail(failure.getReason());
+        });
+    return promise.future();
+  }
+
+  private void defaultMapRecordToAuthority(DataImportEventPayload dataImportEventPayload, MappingMetadataDto mappingMetadata) {
+    try {
+      HashMap<String, String> context = dataImportEventPayload.getContext();
+      var mappingRules = new JsonObject(mappingMetadata.getMappingRules());
+      var parsedRecord = new JsonObject((String) new JsonObject(context.get(MARC_AUTHORITY.value()))
+          .mapTo(Record.class).getParsedRecord().getContent());
+      var mappingParameters = Json.decodeValue(mappingMetadata.getMappingParams(), MappingParameters.class);
+      RecordMapper<Authority> recordMapper = RecordMapperBuilder.buildMapper(MARC_AUTHORITY.value());
+      var authority = recordMapper.mapRecord(parsedRecord, mappingParameters, mappingRules);
+      dataImportEventPayload.getContext().put(AUTHORITY.value(), Json.encode(JsonObject.mapFrom(authority)));
+    } catch (Exception e) {
+      LOGGER.error("Failed to map Record to Authority", e);
+      throw new JsonMappingException("Error in default mapper.", e);
+    }
+  }
+
+  @Override
+  public boolean isEligible(DataImportEventPayload dataImportEventPayload) {
+    if (dataImportEventPayload.getCurrentNode() != null && dataImportEventPayload.getContext().get(MARC_AUTHORITY.value()) != null
+        && ACTION_PROFILE == dataImportEventPayload.getCurrentNode().getContentType()) {
+      var actionProfile = JsonObject.mapFrom(dataImportEventPayload.getCurrentNode().getContent()).mapTo(ActionProfile.class);
+      return ActionProfile.Action.CREATE == actionProfile.getAction() && AUTHORITY == actionProfile.getFolioRecord();
+    }
+    return false;
+  }
+
+  @Override
+  public boolean isPostProcessingNeeded() {
+    return false;
+  }
+
+  private JsonObject prepareAuthority(DataImportEventPayload dataImportEventPayload) {
+    var authorityAsJson = new JsonObject(dataImportEventPayload.getContext().get(AUTHORITY.value()));
+    if (authorityAsJson.getJsonObject(AUTHORITY_PATH) != null) {
+      authorityAsJson = authorityAsJson.getJsonObject(AUTHORITY_PATH);
+    }
+    //Uncomment when
+//    authorityAsJson.put(SOURCE_KEY, MARC_FORMAT);
```

Comment:
When we should uncomment it? Do we have a story for this?

Reply:
yes MODINVSTOR-825, I will update the comment with ticket number

Reply:
But the ticket that you mentioned is for mod-inventory-storage. If we wait for the MODINVSTOR-825 competition, we also should have a mod-inventory ticket to uncomment this part of code.

Reply:
created an issue to not forgot about uncommenting this line https://issues.folio.org/browse/MODINV-576